azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.aad;
+
+import com.microsoft.aad.adal4j.ClientCredential;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserRequest;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+
+import javax.naming.ServiceUnavailableException;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+@Configuration
+@ConditionalOnWebApplication
```

Comment:
Do you want to accept all web application kinds? Does it work with WebFlux?

Reply:
Basically I think we want to accept both servlet and reactive, but I am not sure will the authentication with WebFlux diffs with non reactive web app. @jgrandja

Reply:
@rwinch Recently added WebFlux support for OAuth 2.0 Login so it will work. The [sample](https://github.com/spring-projects/spring-security/blob/master/samples/boot/oauth2login-webflux/src/main/java/sample/web/OAuth2LoginController.java) looks the same. However, the Boot auto-configuration support for OAuth 2.0 Login WebFlux still needs to be added. At the moment the auto-configuration is in the [sample](https://github.com/spring-projects/spring-security/tree/master/samples/boot/oauth2login-webflux/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/client).